.github/workflows/bundleTest-thor.md

```diff
@@ -0,0 +1,353 @@
+- **generate-Zap** : Indicates if ZAP reports are needed for specific test cases. The default is an empty string.You can either pass the name of the test file with .ecl extension or * at the end. The later is used to refer the workuint name. Eg. "KMeansValidateOBT-240531-133100" can be refered as "KMeansValidateOBT*" 
+Example:   
+  Caller Workflow: [build-vcpkg.yml](https://github.com/hpcc-systems/HPCC-Platform/blob/master/.github/workflows/build-vcpkg.yml)
+    ```yaml
+    test-bundles-on-thor-ubuntu-22_04:
+        if: ${{ contains('pull_request,push', github.event_name) }}
+        needs: build-docker-ubuntu-22_04
+        uses: ./.github/workflows/bundleTest-thor.yml
+        with:
+          os: ubuntu-22.04
+          asset-name: 'docker-ubuntu-22_04'
+          generate-Zap: "'KMeansValidateOBT.ecl SVTest*'"
+        secrets: inherit
+    ```
+  **NOTE:** The double quotes are required if * notation is used to specify the name of the test. It makes sure that string function properly during the workflow run.
+- **test-core-file-generation** :A boolean input that determines whether the core file generation test step should be executed. It helps us to know whether the core files are generated as expected. 
+
+## Environmental Variables: 
+```yaml
+env:
+  ML_SUPPRESS_WARNING_FILES: "RegressionTestModified.ecl ClassificationTestModified.ecl"
+  ML_EXCLUDE_FILES: "--ef ClassicTestModified.ecl,SVCTest.ecl,ClassificationTestModified.ecl"
+  BUNDLES_TO_TEST: "ML_Core PBblas GLM  GNN DBSCAN LearningTrees TextVectors KMeans SupportVectorMachines LinearRegression LogisticRegression"       
+  uploadArtifact: false 
+```
+- `ML_SUPPRESS_WARNING_FILES:` Specifies the files requiring a warning suppression parameter during runtime.
+- `ML_EXCLUDE_FILES:` The files specified here are excluded during the run.
+- `BUNDLES_TO_TEST:` Lists the bundles to test.
+- `uploadArtifact:` Determines whether the logs Artifact should be uploaded or not.
+## Steps Involved:
+The steps in the workflow run on the specified operating system, with Ubuntu-22.04 as the default.
+- **Download Package**  
+This step enables us to download the ready-to-install HPCC Platform's artifact built on the latest commit.
+  ```yaml
+      - name: Download Package
+          uses: actions/download-artifact@v3
+          with:
+            name: ${{ inputs.asset-name }}
+            path: ${{ inputs.asset-name }}
+  ```
+- **Install Dependencies**  
+Install the dependencies that are necessary for the platform to install and start sucessfully.
+  ```yaml
+  - name: Install Dependencies
+      shell: "bash"
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y \
+          git \
+          wget \
+          net-tools \
+          tzdata \
+          unzip \
+          xvfb \
+          libxi6 \
+          default-jdk \
+          gdb \
+          ${{ inputs.dependencies }}
+  ```
+- **Install Package**  
+The HPCC Platform is installed from the downloaded artifact.
```

Comment:
I think it would be better if "Install the HPCC Platform from the downloaded artifact, set permissions an configure the Thor engine to use 2 slaves."